src/rest_framework_dso/fields.py

```diff
@@ -19,12 +23,32 @@ def parse_request_fields(fields: Optional[str]):
     return fields.split(",")
 
 
+def get_embedded_field_class(
+    model_field: Union[RelatedField, LooseRelationField, ForeignObjectRel]
+) -> type[AbstractEmbeddedField]:
+    """Return the embedded field type that is suited for a particular model field."""
+    if isinstance(model_field, ForeignObjectRel):
+        # Reverse relations
+        return EmbeddedManyToOneRelField
+    else:
+        if model_field.one_to_many or model_field.many_to_many:
+            return EmbeddedManyToManyField
+        else:
+            return EmbeddedField
```

Comment:
```suggestion
def get_embedded_field_class(
    model_field: Union[RelatedField, LooseRelationField, ForeignObjectRel]
) -> type[AbstractEmbeddedField]:
    """Return the embedded field type that is suited for a particular model field."""
    if isinstance(model_field, ForeignObjectRel):
        # Reverse relations
        return EmbeddedManyToOneRelField
    elif model_field.one_to_many or model_field.many_to_many:
        return EmbeddedManyToManyField
    else:
        return EmbeddedField
```

Reply:
Good point, but tricky. I've updated the code to explicitly show why I'd taken this if/else path.
The reverse M2M likely needs to become a separate type.